src/test/regress/expected/multi_router_planner.out

```diff
@@ -428,13 +428,55 @@ WITH RECURSIVE hierarchy as (
 				ce.company_id = 2))
 SELECT * FROM hierarchy WHERE LEVEL <= 2;
 ERROR:  recursive CTEs are not supported in distributed queries
--- CTE with queries other than SELECT is not supported
+-- Test router modifying CTEs
```

Comment:
We actually still have the `ErrorIfQueryHasModifyingCTE` function in the router planner. Is this something we could get rid of such that, when possible, we'd fully push these queries down instead of doing recursive planning?

Reply:
(we might have to make an exception for INSERTs)

Reply:
I think we can do this, but it requires more stuff than just removing some error checks. I prefer this to be a separate project. What do you think?

Reply:
Makes sense.